README
wxRemind

   wxRemind is a graphical front-end to Remind, a remarkably sophisticated
   calendar and alarm system. wxRemind is similar to Wyrd but is based on
   wxPython rather than curses. The display features a calendar and daily
   event list suitable for visualizing your schedule at a glance. Dates and
   associated events can be quickly selected either with the mouse or cursor
   keys, and dates in the calendar are color coded to reflect the total
   duration of scheduled events. wxRemind integrates with an external editor
   of your choice to make editing of reminder files more efficient, provides
   hotkeys to quickly access the most common Remind options, allows popup,
   sound and/or spoken alerts and can display a postscript calendar of the
   selected month suitable for printing.

  Requirements

   The latest version of wxRemind should be available at wxRemind.

   You will need to have remind and python >= 2.3 and, additionally, wxPython
   >= 2.6.1. For me (Centos 4 and python 2.3), installing wxPython required:

     * wxPython2.6-common-gtk2-unicode-2.6.3.0-fc2_py2.3.i386.rpm
     * wxPython2.6-gtk2-unicode-2.6.3.0-fc2_py2.3.i386.rpm

   Note: analogclock is a recent addition to wxPython (2.6.3) and is required
   for wxRemind. A copy has been included in this distribution under the
   wxWidgets license for the convenience of those with earlier versions of
   wxPython already installed.

   Displaying monthly postscript calendars requires ggv (gnome ghostview).

   Audible, spoken-message alerts require Festival.

  Installation Alternatives

   Follow one of the following alternatives.

    1) Using INSTALL

   Unpack the contents of the wxRemind.tgz in a convenient location, open a
   terminal window and then at a command prompt (not as root but as
   yourself):

   $ cd <path to wxRemind directory>

   $ ./INSTALL

   and follow the instructions.

   This script does the following:

    1. Checks to make sure that you have sufficiently recent versions of
       python and wxPython.
    2. Prompts for a directory in your path and then creates symbolic links
       for the wxRemind executables (wxremind, wxremalert and wxremdata) in
       that directory. The link to wxremalert is necessary for remind to be
       able to use it for alerts. The other links are for convenience.
    3. Provides instructions for using FLOAT and WXREMINDRC to complete the
       installation.

   Using INSTALL has the following advantages:

    1. It doesn't require root privileges.
    2. It leaves your python tree untouched. To remove wxRemind simply delete
       the directory where it was unpacked and the three symbolic links
       created by INSTALL.
    3. It runs on a wide variety of platforms without tweaking. Since the
       package files are in a subdirectory of the root directory where the
       scripts are located, the scripts can automatically find the package
       files ON ANY PLATFORM without further intervention.

   Proceed to Post Installation.

    2) Using setup.py (usually requires root privileges)

   Unpack the contents of the wxRemind.tgz in a convenient location, open a
   terminal window and then at the command prompt (as root):

   # cd <path to wxRemind directory>

   # python setup.py install

   This will create an egg (a compressed zip file) and install it in the
   python tree, usually under site-packages, and install the scripts in your
   system path, usually in /usr/bin/.

   Finally, at a command prompt not as root but as yourself:

   $ wxremsetup

   and follow the instructions. This script checks to make sure you have
   sufficiently recent versions of python and wxPython, installs two files,
   FLOAT and WXREMINDRC, in a temporary location (typically under
   ~/.python-eggs/) and provides instructions about using these files to
   complete the installation.

   Proceed to Post Installation.

    3) Using Easy Install (usually requires root privileges)

   You do not need to download the tarball if you use this alternative.

   Download ez_setup.py, and run it; this will download and install the
   appropriate setuptools egg for your Python version. An easy_install script
   will be installed in the normal location for Python scripts on your
   platform.

   Then at the command prompt (as root):

   # easy_install wxRemind

   This will download and install the egg (a compressed zip file) in the
   python tree, usually under site-packages, and install the scripts in your
   system path, usually in /usr/bin.

   Finally, at a command prompt not as root but as yourself:

   $ wxremsetup

   and follow the instructions. This script checks to make sure you have
   sufficiently recent versions of python and wxPython, installs two files,
   FLOAT and WXREMINDRC, in a temporary location (typically under
   ~/.python-eggs/) and provides instructions about using these files to
   complete the installation.

   Proceed to Post Installation.

  Post Installation

   The last step is to follow the instructions either from INSTALL if using
   approach (1) or from wxremsetup if using either (2) or (3). In either case
   you will need to:

   1) Copy FLOAT to a convenient location and then insert the following line
   somewhere near the beginning of your reminders file (typically
   ~/.reminders) after replacing <Path to FLOAT> with the correct location:

     INCLUDE <Path to FLOAT>/FLOAT

   2) Edit WXREMINDRC to suit your own preferences and then save the
   resulting file as ~/.wxremindrc. Pay particular attention to the settings
   for editor (initially /usr/local/bin/gvim) and reminders (initially
   ~/.reminders). Both of these settings must be correct for your location.

   You can now run wxRemind by opening a terminal window and entering

   $ wxremind

   Once wxRemind is running, pressing ? will bring up a display of usage
   information.

  wxRemind (python package) Files

   wxRemAbout.py
           About page.

   wxRemAlert.py
           Used by default to produce popup/sound alerts for reminders
           triggered by remind. The default behavior can be set in
           wxRemConfig.py.

   wxRemConfig.py
           Color and other customizations. User specific customizations in
           ~/.wxremindrc override the settings in this file.

   wxRemData.py
           Provides the interface to remind.

   wxRemEdit.py
           Dialogs for creating new reminders.

   wxRemHelp.py
           Help page.

   wxRemHints.py
           Hints for creating reminders.

   wxRemind.py
           provides the main GUI interface.

   wxRemSplash.py
           Splash page.

   wxRemVersion.py
           wxRemind version information.

  wxRemind scripts

   wxremind
           Used to start wxRemind at the command prompt. Takes no arguments.

   wxremdata
           Returns day schedule as console output. Optional YYYY MM DD date
           argument, defaults to today.

   wxremalert
           Produces an alert. Details in wxRemConfig.py. Can be used
           independently of wxRemind and remind.

   INSTALL, wxremsetup
           Used for dependency checks and initial post installation
           configuration.

  License

   Copyright (c) 2006 Daniel A. Graham <daniel.graham@duke.edu>

   This program is free software; you can redistribute it and/or modify it
   under the terms of the GNU General Public License as published by the Free
   Software Foundation (www.gnu.org/licenses/gpl.html); either version 2 of
   the License, or (at your option) any later version. This program is
   provided in the hope that it will be useful, but without any warranty;
   without even the implied warranty of merchantability or fitness for a
   particular purpose. Details are provided in the included COPYING file.


CHANGES
0.5.0

    Significant cosmetic improvements.

    Replaced listctrl header and statusbar with textcrtl's to solve
    background color problems and to allow selected date to be
    centered. Switched from 1 to 3 columns for event list to allow the
    use of non-monospaced fonts.

    Added configuration option for selecting font sizes, and for
    selecting background colors either by 'family' or individually.

    Selection automatically removed from event when focus switches
    from event list to calendar.

0.4.3

    Added wxremsetup for post installation configuration using
    either setup.py or easy_install.  Checks python and wxPython
    dependencies (never got the latter to work with
    install_requires), unpacks FLOAT and WXREMINDRC to a temporary
    directory and provides instructions similar to INSTALL for using
    these files to finish the installation.

    Edited index.html and README to reflect these changes.

0.4.2

    Initial attempt at setup.py configuration with access to FLOAT
    and WXREMINDRC.

0.4.1

    Added CHANGES, COPYING, FLOAT, README and WXREMINDRC to egg.

0.4 2006-05-19

    Changed version numbering to traditional format.

    Added setup.py (based on setuptools/ez_setup --- not distutils)
    and easy_install to INSTALL as optional installation methods.

2006-05-18

    Replaced calls to 'rem' with direct calls to 'remind' for the
    convenience of those whose remind installations did not include
    rem. 

    Fixed bug in specifying the default location of the reminders
    file.

    Added a 'new reminder' dialog to facilitate the process of
    creating reminders to those new to remind. Untimed, timed, alert
    and floating reminders are supported. 

    Added a page showing hints for creating new reminders.

    Added an optional status bar display of the reminder
    corresponding to the selected event --- displays the complete
    details of the selected event without the necessity of opening
    the external editor.

    When the event list takes focus, the first item is now selected
    unless another item was already selected.

    Added border styles (sunken, flat, raised) for event list,
    calendar and status bar to the configuration options.

    Removed leading zero on day in searches.

2006-05-14

    Converted INSTALL to a python script.  Better dependency checks
    and more user friendly location choice for symbolic links. Does
    not require root permissions, leaves the python tree untouched
    and should work on all platforms since the package files are
    located in a subdirectory of the directory containing the
    scripts.

    User specific customizations can now be set in ~/.wxremindrc.

2006-05-07

    New alert feature:  Reminders can now 'RUN' wxRemAlert.py
    (provided) to display a pop-up message and/or play a sound file
    or, if available, invoke festival (Festival Speech Synthesis) to
    read the event message aloud. Details and an example are
    provided in wxRemConfig.py (provided).

    New PS monthly calendar: display and optionally print a
    postscript calendar for selected month.  Uses remind's rem2ps
    and requires ggv (Gnome GhostView) for display.

    New font configuration option: font facenames and sizes for today
    button and event list, 

    New time display option: choose between twelve-hour (AM/PM) and
    twenty-four hour time displays.

    Improved, wyrd-like configuration for editor and remind strings
    (untimed, timed, etc.)

    Improved syntax for floating reminders: f(y,m,d,n) now sets a
    reminder due on y/m/d with an n day advance warning - see the
    documentation in FLOAT (provided).

    Cosmetic: Removed leading zeros from hours and days.

2006-05-02

    Initial release

vim: ft=text


wxRemind/wxRemSetup.py
#!/usr/bin/python
# $Id: wxRemAlert.py 14 2006-05-09 12:20:50Z dag $
import wx
# from wxRemConfig import * 
import sys, os, re
import shutil
msg = []
warn = []

d = {'remind' : '', 'festival' : '', 'ggv' : ''}

reminders = os.path.expanduser("~/.reminders")
has_reminders = os.path.isfile(reminders)
reminders_backup = os.path.expanduser("~/.reminders.bak")
d['reminders'] = reminders

wxremindrc = os.path.expanduser("~/.wxremindrc")
has_wxremindrc = os.path.isfile(wxremindrc)
new_wxremindrc = False

wxremfloat = os.path.expanduser("~/.wxremfloat")
has_wxremfloat = os.path.isfile(wxremfloat)

include_line = "INCLUDE %s" % wxremfloat

alert_wave = '/usr/share/sounds/KDE_Notify.wav'
has_sound = os.path.isfile(alert_wave)
if has_sound:
    d['sound'] = alert_wave
else:
    d['sound'] = ''

def make_wxremindrc(d):
    global msg, warn, new_wxremindrc
    WXREMINDRC = """\
# Configuration settings for wxRemind
# This file should be named ~/.wxremindrc

# REMIND 
remind = '%(remind)s'

# Festival (necessary for spoken alerts)
festival = '%(festival)s'

# ggv (Gnome Ghostview - necessary to display and print monthly calendars)
ggv = '%(ggv)s'

# The reminders file
reminders = '%(reminders)s'

# TIME DISPLAY 12 hour (1) or 24 hour (0)
twelvehour = 1

# MONTHLY CALENDAR ORIENTATION landscape (1) or portrait (0)
landscape = 1

# show splashscreen on startup for this many seconds or 0 for not at all
showsplash = 2

# EDITOR - Leave empty to use interal editor
editor = ''
# editor = "/usr/local/bin/gvim"
# editor = "/usr/bin/emacs"


# EDIT COMMAND SUBSTITUTIONS
# These are irrelevant if using the default interal editor
# %%(e)s -> the editor named above
# %%(n)s -> line number to edit
# %%(f)s -> file name
# COMMAND FOR EDITING AN OLD APPOINTMENT
# editold = "%%(e)s -f +%%(n)s %%(f)s"        # gvim 
# editold = "%%(e)s +%%(n)s %%(f)s"         # emacs
# COMMAND FOR EDITING A NEW APPOINTMENT (the last line in the file)
# editnew = "%%(e)s -f + %%(f)s"          # gvim
# editnew = "%%(e)s +999999 %%(f)s"     # emacs

# A TIMED EVENT WITH AN ALERT
# The system command to "play" a wav file
alert_play = '%(play)s'
# The wave file to play
alert_wave = '%(sound)s'
# ALERT DEFAULTS 
# Display the alter box (1) or not (0)
alert_display = 1
# Silent (0), execute alert_soundcmd (1) or play spoken message (2)
alert_sound = 1
# Add a time appropriate greeting to spoken message, e.g., "Good morning"
alert_greeting = 1
# Add a name if non-null to the spoken greeting, e.g, "Good morning, Dan"
alert_whom = ''
# Replace all 3-digit numbers in msg with spoken equivalents, e.g., Replace
# '201' with '2 O1' where O is an uppercase o) and '385' with '3 85'. Thus
# '201' would be spoken as 'two oh one' and '385' as 'three eighty-five'.
# Why course and room numbers are spoken this way is a mystery to me.
alert_parsenums = 1

# For example, with 
#   alert_display = 1
#   alert_sound = 1
#   alert_greeting = 1
#   alert_whom = 'Dan' 
# and reminder:
#   REM 11 Apr 2006 AT 10:50 +5 DUR 1:15 RUN wxremalert -d0 -s2 %%"Economics 201%%" %%1%%
# the calendar for Apr 11 would display
#   10:50AM - 12:05PM  Economics 201
# Then, at 10:45 on Apr 11 the following message would be spoken (-s2 overrides
# the default alert_sound) but not displayed (-d0 overrides the default
# alert_display):
#   "Good morning, Dan. The time is ten forty-five. Economics two oh one five
#    minutes from now."
# and at 10:50 the the following message would be spoken:
#   "Good morning, Dan. The time is ten fifty. Economics two oh one now."s

# FONTSIZES
# The base fontsize
basefontsize = 10
# The adjustment to the base font size for the status bar
statusfontadj = 0
# The adjustment to the base font size for the event list
listfontadj = 0
# The adjustment to the base fontsize for the selected date header above the
# event list
datefontadj = 0
# The adjustment to the base fontsize for today button
buttonfontadj = 0
# The adjustment to the base fontsize for the calendar
calendarfontadj = 0

# Custom fontsizes may require adjustments to panel sizes.
# The absolute startup size (horizontal, vertical) of the event list
eventsize = (500,380)
# The event list resizes automatically with the window size but the calendar
# and clock keep their startup sizes.
# The default size of the calendar takes account of the font size. 
# The following adjustment (horizontal, vertical) fine tunes this default. 
calsizeadj = (16, -4) 
# The absolute size of the (fontless) analogclock: 
clocksize = 190
# The absolute size of the internal editor window:
editorsize = (800,400)

# COLORS
# calendar busy times and colors
busy0 = (30,'SLATEBLUE2')
busy1 = (75, 'BLUE2')
busy2 = (150, 'GREEN4')
busy3 = (225, 'DARKORANGE')
busy4 = (300, 'RED')
# Days with appointments totaling less than 30 minutes have foreground (font)
# color BLACK. Days with at least 30 minutes but less than 75 are SLATEBLUE2
# and so forth.
# Time to assign to events with no duration in computing day event totals:
zerominutes = 15

# for the calendar
holidaycolor = 'BLACK'
headercolor = 'NAVYBLUE'

# BACKGROUND COLORS
# bgcolor: The general background color.
# nfcolor: The background color for the event list and calendar, whichever has
# the focus.
# fcolor:  The background color for the event list and calendar, whichever does
# not have the focus.

# There are two ways of specifying these background colors. The first is simply
# to specify these colors directly, e.g., 
# handcolor = 'GRAY50'
# bgcolor   = "GRAY90"
# nfcolor   = "GRAY94"
# fcolor    = "GRAY99"

# The second way is to specify a color 'family' from one of the 'numbered'
# color families supported by python. Here is a partial list:

#    "AZURE" "BISQUE" "CORNSILK" "DARKOLIVEGREEN" "DARKSEAGREEN" "HONEYDEW"
#    "IVORY" "KHAKI" "LAVENDERBLUSH" "LEMONCHIFFON" "LIGHTGOLDENROD"
#    "LIGHTSKYBLUE" "LIGHTSTEELBLUE" "LIGHTYELLOW" "MISTYROSE" "NAVAJOWHITE"
#    "PALEGREEN" "PEACHPUFF" "THISTLE" "WHEAT" 

# For each of these python supports four named colors, e.g., 'IVORY1',
# 'IVORY2', 'IVORY3' and 'IVORY4'. Simply setting
#    colorfamily = "IVORY"
# has exactly the same effect as setting
#    handcolor = "IVORY4"
#    bgcolor   = "IVORY3"
#    nfcolor   = "IVORY2"
#    fcolor    = "IVORY1"
# and, in fact, will override any individual color settings.

colorfamily = "IVORY"


# borders: sunken (-1), flat (0) or raised (1)
# event and calendar borders
ec_border = 1
""" % d
    if has_wxremindrc:
        fo = open("%s.new" % wxremindrc, 'w')
        msg.append("""\
Found an existing %s
    and created a new wxremindrc as %s.new""" % (wxremindrc, wxremindrc))
        new_wxremindrc = True
    else:
        fo = open(wxremindrc, 'w')
        msg.append("Created new %s" % wxremindrc)
    fo.write(WXREMINDRC)
    fo.close()

def make_wxremfloat():
    global msg, warn
    WXREMFLOAT = """\
##########################################################
# Do Not Edit - Updates to wxRemind overwrite this file. #
##########################################################
# Support for floating reminders
# This file should be named ~/.wxremfloat
#   To enable floating reminders make sure your reminders file 
#   includes the line
#
#      INCLUDE %s
#
#   prior to any use of float()
#
IF  ($CalMode || $PSCal )
    # For Cal and PS calendars, only trigger the event on the due date.
    FSET float(y,m,d,n) trigger(date(y,m,d))
ELSE
    # For Simple Calendar (rem -s) and other modes, also set warning trigger,
    # priority and suffixes.
    FSET float(y,m,d,n) iif(date(y,m,d) == today(), trigger(today()) + \\
    " PRIORITY 1000", trigger(MAX(realtoday(), date(y,m,d)-n)) + \\
    iif(n >= date(y,m,d) - today(), " PRIORITY " + \\
    (1000 + (date(y,m,d) - today())),"")))

    FSET msgsuffix(x) iif(0 > x, "", \
    998 >= x, " (" + (1000-x) + " days ago)", \\
    999 >= x, " (yesterday)", \\
    1000 >= x, " (today)", \\
    1001 >= x, " (tomorrow)", \\
    2000 >= x, " (in " + (x - 1000) + " days)", "")

    # This calsuffix is used by wxRemind
    FSET calsuffix(x) iif(0 > x, "", \\
    998 >= x, " (" + (1000-x) + "  days ago)", \\
    999 >= x, " (yesterday)", \\
    1000 >= x, " (today) ", \\
    1001 >= x, " (tomorrow)", \\ 
    2000 >= x, " (in " + (x - 1000) + " days)", "")
ENDIF
""" % wxremfloat
    fo = open(wxremfloat, 'w')
    fo.write(WXREMFLOAT)
    fo.close()
    msg.append("Created %s" % wxremfloat)

def PathSearch(filename):
    search_path = os.getenv('PATH').split(os.pathsep)
    for path in search_path:
        candidate = os.path.join(path,filename)
        if os.path.os.path.isfile(candidate):
            return os.path.abspath(candidate)
    return ''

def DepCheck():
    global d, msg, warn
    v = sys.version.split()[0]
    if v < '2.3':
        warn.append("Error: Found Python %s. You need Python >= 2.3 to use this package." % v)
    else:
        msg.append("Found python version %s" % v)
    try:
        import wx
        w = wx.VERSION_STRING
        if w < '2.6.1':
            warn.append("Error: Found wxPython %s. You need wxPython >= 2.6.1 to use this package." % w)
        else:
            msg.append("Found wxPython version %s" % w)
    except:
        warn.append("Error: You do not have wxPython installed. You need wxPython >= 2.6.1 to use this package.")

    d['remind'] = PathSearch('remind')
    if d['remind']:
        msg.append("Found remind at '%s'" % d['remind'])
    else:
        warn.append("""
Error: Could not find remind in your PATH. Remind must be installed 
with the executable 'remind' in your system path.""")

    d['festival'] = PathSearch('festival')
    if d['festival']:
        msg.append("Found festival at '%s'" % d['festival'])
    else:
        warn.append("""\
Error: Could not find festival in your PATH. Festival must be installed
with the executable 'festival' in your system path for you to be able to
use spoken alerts.""")

    d['ggv'] = PathSearch('ggv')
    if d['ggv']:
        msg.append("Found ggv at '%s'" % d['ggv'])
    else:
        warn.append("""\
Error: Could not find ggv in your PATH. Gnome ghostview must be installed
with the executable 'ggv' in your system path for you to be able to display
and print monthly calendars.""")

    d['play'] = PathSearch('play')
    if d['play']:
        msg.append("Found play at '%s'" % d['play'])
    else:
        warn.append("""\
Error: Could not find play in your PATH. Gnome ghostview must be installed
with the executable 'ggv' in your system path for you to be able to display
and print monthly calendars.""")

    if has_sound:
        msg.append("Found alert_wave file: %s" % alert_wave)
    else:
        warn.append("""\
Could not find the default alert sound wave file, %s. You will need to correct
the setting for 'alert_wave' in %s to be able to use sound alerts.""" %
(alert_wave, wxremindrc))


def make_reminders():
    global msg, warn
    # If there is an existing reminders file, see if it includes wxremfloat
    # and, if it does not, backup the file and insert the include line at the
    # beginning.
    if has_reminders:
        found = False
        msg.append("Found reminders file: '%s'" % reminders) 
        regex = re.compile(r'^\s*INCLUDE .*\.wxremfloat')
        fo = open(reminders, 'r')
        lines = fo.readlines()
        fo.close()
        for line in lines:
            if regex.match(line):
                found = True
                break
        if found:
            msg.append("  and it contains the requisite '%s' " % (include_line)) 
        else:
            msg.append("  and it does not contain the requisite '%s'"
                    % (include_line)) 
            lines.insert(0, "%s\n" % include_line)
            shutil.copyfile(reminders, "%s.bak" % reminders)
            msg.append("  Created backup %s.bak"% reminders)
            fo = open(reminders, 'w')
            fo.writelines(lines)
            fo.close()
            msg.append("  Inserted '%s' as the first line in %s" 
                    % (include_line, reminders))
    else:
        fo = open(reminders, 'w')
        fo.write(include_line)
        fo.close()
        msg.append("Created new %s" % reminders)



def report():
    global new_wxremindrc
    if new_wxremindrc:
        new_msg = """\
IMPORTANT: It is strongly recommended that you edit ~/.wxremindrc.new to
reflect any custom settings that you may have made in your existing
~/.wxremindrc, and then save ~/.wxremindrc.new as ~/.wxremindrc.
"""
    else:
        new_msg = ''

    if len(warn) > 0:
        warnings = """
The following warnings were generated during configuration:
  %s""" % '\n  '.join(warn)
    else:
        warnings = ''
    if len(msg) > 0:
        messages = """
The following information messages were generated during configuration:
  %s""" % '\n  '.join(msg)
    else:
        messages = ''

    print """
                ===============================
                    Configuration Completed 
                ===============================
%s%s

%s
Absent warning messages to the contrary, you should now be able run
wxremind and the other wxRemind scripts.

Once wxremind is running, pressing ? will bring up a display of usage
information.
""" % (messages, warnings, new_msg)

def setup():
    print """
                ===============================
                     wxRemind Configuration 
                ===============================

This configuration process will check for sufficiently recent versions of
'python' and 'wxPython', for the presence of 'remind', 'festival', 'ggv' and
'play' in your system path and for the existence of the default alert sound
file. 

It will then create '.wxremfloat' in your home directory, overwriting any
existing file.  The contents of this file will make it possible for you to 
use floating reminders.

Next it will check for the existence of '.reminders' and '.wxremindrc' in your
home directory. If '.reminders' exists it will be checked to make sure that it
contains an 'include .wxremfloat' line. If it does not, first a backup will be
made named '.reminders.bak' and then the appropriate line will be inserted at
the beginning of the original file. If '.reminders' does not exist, then it
will be created with the necessary line as its only content.

A new configuration file will then be created called '.wxremindrc' if this file
does not alread exist and '.wxremindrc.new' otherwise. This file will
automatically contain the correct settings for 'remind', 'festival', 'ggv' and
'play' and, of course, for 'reminders' and 'wxremfloat'.

Finally, a report of messages generated during configuration will be displayed.

      THIS SCRIPT MUST BE RUN WITH YOUR NORMAL USERID AND NOT AS ROOT. 
         CANCEL IF THIS IS NOT THE CASE AND RESTART AS YOURSELF.
"""

    ans = raw_input('Continue with configuration? [yN] ')
    if ans == 'y':
        DepCheck()
        make_wxremindrc(d)
        make_wxremfloat()
        make_reminders()
        report()
    else:
        print """
wxRemind configuration cancelled.
"""

if __name__ == "__main__":
    setup()
